pyvex/block.py

```diff
@@ -83,10 +87,10 @@ def __init__(self, data, mem_addr, arch, max_inst=None, max_bytes=None,
         self.addr = mem_addr
         self.arch = arch
 
-        self.statements = []
-        self.next = None
+        self.statements = [] # type: List[IRStmt]
```

Comment:
This syntax is not necessary any more, post python 3.5 iirc. You can define instance variables at the class level, along with their type. Example:

```py
class Person:
    name: str

    def __init__(self, name: str):
        self.name = name

```

Reply:
That syntax doesn't kick in until py 3.6. It is an error on 3.5 and we still need to support 3.5.

Reply:
I am still targeting 3.5+ and have my PyCharm configured as such that breaking compatibility is an error.
There are a few pitfalls like that and as far as I saw automatic conversion is an option anyway, so I would suggest waiting for at least pypy 3.6 to be stable before dropping 3.5 support.

Reply:
Before merging this, do we want to reconsider since we've now moved to targeting 3.6?

Reply:
At this point I would just really like this to merged asap so I don't have to rebase all the time and use my own angr fork for other things I am working on that require type annotations.

 I am not sure if this makes any functional difference or has any advantage. https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations sounds like it might allow to declare that a variable must be initialized after `__init__` while allowing it to be set to `None` at some point during `__init__`. I think I had this issue in some places, so if this solves this I am definitely in favor of changing that.